Sources/KlaviyoSwift/KlaviyoEnvironment.swift

```diff
@@ -9,18 +9,33 @@ import Foundation
 
 var environment = KlaviyoEnvironment.production
 
+let PRODUCTION_HOST = "https://a.klaviyo.com"
+let encoder = {
+    let encoder = JSONEncoder()
+    encoder.keyEncodingStrategy = .convertToSnakeCase
+    encoder.dateEncodingStrategy = .iso8601
+    return encoder
+}()
+
 struct KlaviyoEnvironment {
     var archiverClient: ArchiverClient
     var fileClient: FileClient
     var data: (URL) throws -> Data
     var logger: LoggerClient
     var networkSession: NetworkSession
+    var apiURL: String
+    var encodeJSON: (Encodable) throws -> Data
+    var uuid: () -> UUID
+    var date: () -> Date
```

Comment:
not to make this more complex but seems it would be worth refactoring into a new struct or something like `RequestHelper` and have one of those in the Environment struct, it seems strange to have a date lambda on the environment

Reply:
Well there might be a lot of cases where you want the current date beyond building requests...

Reply:
The idea is to control it in the test environment.

Reply:
i just find that its a little strange to have this if its mainly for the test environment, doesnt feel like a well integrated solution

Reply:
I added a sub environment.